### PR TITLE
librbd: allow disabling journaling for snapshot based mirroring image

### PR DESCRIPTION
**Pull request by trociny**
Fixes: https://tracker.ceph.com/issues/49282
Signed-off-by: Mykola Golub <mgolub@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/librbd/operation/DisableFeaturesRequest.cc [hunk 1]
```diff
@@ -306,15 +306,20 @@ Context *DisableFeaturesRequest<I>::handle_get_mirror_image(int *result) {
     return handle_finish(*result);
   }
 
-  if ((mirror_image.state == cls::rbd::MIRROR_IMAGE_STATE_ENABLED) && !m_force) {
-    lderr(cct) << "cannot disable journaling: image mirroring "
+  if (mirror_image.state == cls::rbd::MIRROR_IMAGE_STATE_ENABLED &&
+      mirror_image.mode == cls::rbd::MIRROR_IMAGE_MODE_JOURNAL && !m_force) {
+    lderr(cct) << "cannot disable journaling: image journaling mirroring "
```
Comment (dillaman) [inline src/librbd/operation/DisableFeaturesRequest.cc:311]: Nit:
```suggestion
    lderr(cct) << "cannot disable journaling: journal-based mirroring "
```

**Review by dillaman — APPROVED** [review]
Otherwise, LGTM

**Comment by trociny** [comment]
thanks. updated.

**Comment by dillaman** [comment]
jenkins test make check

**Pull request merged**